github/create_deployment.yaml

```diff
@@ -0,0 +1,109 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: github-create-deployment
+  description: |
+    This Task will create a GitHub deployment for a repository.
+spec:
+  params:
+    - name: GITHUB_HOST_URL
+      description: |
+        The GitHub host, adjust this if you run a GitHub enterprise.
+      default: "api.github.com"
+    - name: REPO_FULL_NAME
+      description: |
+        The GitHub repository full name, i.e: tektoncd/catalog.
+    - name: REF
+      description: |
+        The ref to deploy. This can be a branch, tag, or SHA.
+    - name: TASK
+      description: |
+        Specifies a task to execute (e.g., deploy or deploy:migrations).
+      default: "deploy"
+    - name: AUTO_MERGE
+      description: |
+        Attempts to automatically merge the default branch into the
+        requested ref, if it's behind the default branch.
+      default: "true"
+    - name: REQUIRED_CONTEXTS
+      description: |
+        The status contexts to verify against commit status checks. To
+        verify all unique contexts before creating a deployment, pass
+        in "*". To bypass checking entirely, pass an empty string.
+      default: "*"
+    - name: PAYLOAD
+      description: |
+        JSON payload with extra information about the deployment.
+      default: ""
+    - name: ENVIRONMENT
+      description: |
+        Name for the target deployment environment (e.g., production, staging, qa).
+    - name: DESCRIPTION
+      description: |
+        Short description of the deployment.
+      default: ""
+  steps:
+    - name: create-deployment
+      env:
+        - name: GITHUBTOKEN
+          valueFrom:
+            secretKeyRef:
+              name: github
+              key: token
+
+      image: registry.access.redhat.com/ubi8/ubi:latest
+      script: |
+        #!/usr/libexec/platform-python
+        import json
+        import os
+        import http.client
+
+        deployment_url = "/repos/$(params.REPO_FULL_NAME)/deployments"
+
+        if "$(params.REQUIRED_CONTEXTS)" == "*":
+            required_contexts = None
+        elif "$(params.REQUIRED_CONTEXTS)" == "":
+            required_contexts = []
+        else:
+            required_contexts = "$(params.REQUIRED_CONTEXTS)".split(",")
```

Comment:
This is a little bit janky, as there are 3 cases for the required_contexts provided by the GitHub API itself.

- You want to verify all unique contexts (this is the default in the API, and requires omitting the field)
- You want to bypass checks entirely (this requires providing an empty array)
- You provide an array of status contexts you want to verify

So in the code:

- Use * to assign None as a way of omitting the field
- Use empty string as an alias for an empty array, otherwise
- Create a comma-separated list from a string (eg. `ci/circleci, cla/google`)

Open to any better ideas here as this was the best I could think of!